***
# *What types of films are currently doing the best at the box office and what are some actionable insights from that data that can be used when starting a new movie studio?*
***

# imports and initialization

In [2]:
import pandas as pd

In [2]:
ls ./datasets

bom.movie_gross.csv*       imdb.title.ratings.csv*
imdb.name.basics.csv*      rt.movie_info.tsv*
imdb.title.akas.csv*       rt.reviews.tsv*
imdb.title.basics.csv*     tmdb.movies.csv*
imdb.title.crew.csv*       tn.movie_budgets.csv*
imdb.title.principals.csv*


# bom.movie_gross.csv

## Summary

Table with movie titles, a studio abbreviation, domestic gross, foreign gross and year.

There are 3387 movies, but 40% of them are missing their foreign_gross, plus a minor amount don't have a domestic gross.  At least some of the movies with no listed domestic gross did not have a domestic release.

## Columns:  
* title
* studio
* domestic_gross
* foreign_gross
* year

## EDA

In [8]:
bom_df = pd.read_csv('datasets/bom.movie_gross.csv')

In [9]:
bom_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [10]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null object
year              3387 non-null int64
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [11]:
# remove the commas from the data in the foreign_gross column and convert it to float
bom_df['foreign_gross'] = bom_df['foreign_gross'].str.replace(',', '')

bom_df['foreign_gross'] = bom_df['foreign_gross'].astype('float')

In [12]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null float64
year              3387 non-null int64
dtypes: float64(2), int64(1), object(2)
memory usage: 132.4+ KB


why are there NAN in the foreign gross column?

is it because there is no foreign gross or because there is no data?

there are also some rows with no domestic gross, why?

In [13]:
bom_df.describe()

domestic_gross  foreign_gross         year
count    3.359000e+03   2.037000e+03  3387.000000
mean     2.874585e+07   7.487281e+07  2013.958075
std      6.698250e+07   1.374106e+08     2.478141
min      1.000000e+02   6.000000e+02  2010.000000
25%      1.200000e+05   3.700000e+06  2012.000000
50%      1.400000e+06   1.870000e+07  2014.000000
75%      2.790000e+07   7.490000e+07  2016.000000
max      9.367000e+08   9.605000e+08  2018.000000

In [70]:
# first lets look at the data with no foreign_gross
bom_df[bom_df['foreign_gross'].isna()]

title      studio  domestic_gross  \
222                                  Flipped          WB       1800000.0   
254   The Polar Express (IMAX re-issue 2010)          WB        673000.0   
267                           Tiny Furniture         IFC        392000.0   
269            Grease (Sing-a-Long re-issue)        Par.        366000.0   
280                          Last Train Home       Zeit.        288000.0   
...                                      ...         ...             ...   
3382                               The Quake       Magn.          6200.0   
3383             Edward II (2018 re-release)          FM          4800.0   
3384                                El Pacto        Sony          2500.0   
3385                                The Swan  Synergetic          2400.0   
3386                       An Actor Prepares       Grav.          1700.0   

      foreign_gross  year  
222             NaN  2010  
254             NaN  2010  
267             NaN  2010  
269             NaN  2010  
280             NaN  2010  
...             ...   ...  
3382            NaN  2018  
3383            NaN  2018  
3384            NaN  2018  
3385            NaN  2018  
3386            NaN  2018  

[1350 rows x 5 columns]

doing a google search on Flipped, I found that there is an international box office amount of $2,534,781 which is greater than the domestic_gross

listed at https://www.the-numbers.com/movie/Flipped#tab=summary

so foreign_gross info is available just not in this dataset

is it worth trying to scrape it? 1350 is 40 percent of the dataset, so it might be worth the effort

the data at www.the-numbers.com is not easily scraped and is in fact prohibited, but they do offer an academic use dataset.  Maybe it could be useful.

In [73]:
# next lets look at the movies with no domestic_gross and sort them
for_sorted_bom_df = bom_df[bom_df['domestic_gross'].isna()].sort_values(by=['foreign_gross'], ascending=False)

In [74]:
for_sorted_bom_df

title   studio  domestic_gross  \
2825                       Secret Superstar      NaN             NaN   
2392                    Finding Mr. Right 2       CL             NaN   
1975         Surprise - Journey To The West       AR             NaN   
2468                                 Solace      LGP             NaN   
1268                             22 Bullets    Cdgm.             NaN   
871                            Solomon Kane     RTWC             NaN   
1659                             Jessabelle      LGF             NaN   
1308             Matru Ki Bijlee Ka Mandola      FIP             NaN   
928                            The Tall Man    Imag.             NaN   
537                                   Force     FoxS             NaN   
933                   Keith Lemon: The Film      NaN             NaN   
936                     Lula, Son of Brazil     NYer             NaN   
1681                              14 Blades     RTWC             NaN   
1685        Jack and the Cuckoo-Clock Heart   Shout!             NaN   
1340                      The Snitch Cartel       PI             NaN   
1342           All the Boys Love Mandy Lane     RTWC             NaN   
966                          The Cup (2012)     Myr.             NaN   
230              It's a Wonderful Afterlife      UTV             NaN   
1739                              Lila Lila    Crnth             NaN   
1368                                6 Souls     RTWC             NaN   
2595                                  Viral   W/Dim.             NaN   
1017                              Dark Tide      WHE             NaN   
298   Celine: Through the Eyes of the World     Sony             NaN   
302                              White Lion    Scre.             NaN   
1079                         The Green Wave       RF             NaN   
306                        Badmaash Company     Yash             NaN   
713                        Empire of Silver     NeoC             NaN   
327                      Aashayein (Wishes)  Relbig.             NaN   

      foreign_gross  year  
2825    122000000.0  2017  
2392    114700000.0  2016  
1975     49600000.0  2015  
2468     22400000.0  2016  
1268     21300000.0  2013  
871      19600000.0  2012  
1659      7000000.0  2014  
1308      6000000.0  2013  
928       5200000.0  2012  
537       4800000.0  2011  
933       4000000.0  2012  
936       3800000.0  2012  
1681      3800000.0  2014  
1685      3400000.0  2014  
1340      2100000.0  2013  
1342      1900000.0  2013  
966       1800000.0  2012  
230       1300000.0  2010  
1739      1100000.0  2014  
1368       852000.0  2013  
2595       552000.0  2016  
1017       432000.0  2012  
298        119000.0  2010  
302         99600.0  2010  
1079        70100.0  2012  
306         64400.0  2010  
713         19000.0  2011  
327          3800.0  2010

the highest grossing movie on this list, Secret Superstar, did not have a domestic_gross

and neither did the 2nd, Finding Mr. Right 2.

# imdb.name.basics.csv

## Summary

Table with names of people working in the movie industry, their primary professions and the titles they are known for.  It includes an id for that person, and movie id's that can that can be referenced in other IMDB tables.

## Columns
* nconst
* primary_name
* birth_year
* death_year
* primary_profession
* known_for_titles

## EDA

### primary_profession and known_for_titles both contain lists.
* primary_profession contains a list of strings of job titles
* know_for_titles contains a list of presumably foreign keys

### todo: split the primary_profession column in some way tbd

In [4]:
imdb_name_basics_df = pd.read_csv('datasets/imdb.name.basics.csv')

In [14]:
imdb_name_basics_df.head(10)

nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   
5  nm0062879     Ruel S. Bayani         NaN         NaN   
6  nm0063198              Bayou         NaN         NaN   
7  nm0063432      Stevie Be-Zet         NaN         NaN   
8  nm0063618          Jeff Beal      1963.0         NaN   
9  nm0063750    Lindsay Beamish         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   
5         director,production_manager,miscellaneous   
6                                             actor   
7                               composer,soundtrack   
8              composer,music_department,soundtrack   
9                             actress,miscellaneous   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256  
5  tt2590280,tt0352080,tt0216559,tt2057445  
6                      tt6579724,tt0093116  
7  tt3106212,tt0478239,tt0264917,tt1626606  
8  tt0183659,tt2545118,tt0384766,tt1856010  
9  tt0404826,tt0111756,tt0367027,tt1492842

In [6]:
imdb_name_basics_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 606648 entries, 0 to 606647
Data columns (total 6 columns):
nconst                606648 non-null object
primary_name          606648 non-null object
birth_year            82736 non-null float64
death_year            6783 non-null float64
primary_profession    555308 non-null object
known_for_titles      576444 non-null object
dtypes: float64(2), object(4)
memory usage: 27.8+ MB


In [24]:
# look into the rows with actors
# first need to filter out the rows with NaN for primary_profession
no_pp_nan_imdb_name_basics_df = imdb_name_basics_df[imdb_name_basics_df['primary_profession'].notna()] 

In [27]:
# then filter looking for rows that contain a primary_profession of actor
no_pp_nan_imdb_name_basics_df[no_pp_nan_imdb_name_basics_df['primary_profession'].str.contains('actor')]

nconst           primary_name  birth_year  death_year  \
2       nm0062070             Bruce Baum         NaN         NaN   
6       nm0063198                  Bayou         NaN         NaN   
14      nm0066093           Ahmad Bedair      1945.0         NaN   
15      nm0066163         Arnaud Bedouët         NaN         NaN   
17      nm0066897  Mohammad-Ali Behboudi      1956.0         NaN   
...           ...                    ...         ...         ...   
606633  nm9985838      Anthony Simonetti         NaN         NaN   
606637  nm9989005           Jae-Hwan Kim         NaN         NaN   
606639  nm9989840     Mariya Shekaturova         NaN         NaN   
606641  nm9990047         Joseph Jackson         NaN         NaN   
606647  nm9993380         Pegasus Envoyé         NaN         NaN   

                primary_profession                         known_for_titles  
2       miscellaneous,actor,writer  tt1470654,tt0363631,tt0104030,tt0102898  
6                            actor                      tt6579724,tt0093116  
14             actor,miscellaneous  tt2290391,tt0424830,tt0297398,tt2338117  
15           actor,writer,director  tt1355327,tt1977735,tt0173525,tt1710520  
17                           actor  tt6836462,tt4591834,tt0375911,tt2101473  
...                            ...                                      ...  
606633                       actor                                tt8028908  
606637                       actor                                      NaN  
606639                       actor                                tt5953804  
606641                       actor                                      NaN  
606647       director,actor,writer                                tt8743182  

[177838 rows x 6 columns]

In [28]:
# birth_year and death_year are not relevent because there are so many missing values

# imdb.title.akas.csv

## Summary

A table of the multiple names a movie may have around the world that all refer to a single id.

## Columns

* title_id 
  * is a shared key for the imdb tables
* ordering
  * an index relative to each title
* region
  * a country code
* language 
  * may be the dubbed language of the film or maybe subtitles
* attributes
  * extra notes, most don't have them
* is_original_title 
  * is either 0 or 1 (some entries that are the same as the original title are 0 for some reason)

## EDA

In [30]:
imdb_title_akas_df = pd.read_csv('datasets/imdb.title.akas.csv')

In [35]:
imdb_title_akas_df.head(20)

title_id  ordering                                    title region  \
0   tt0369610        10                            Джурасик свят     BG   
1   tt0369610        11                        Jurashikku warudo     JP   
2   tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3   tt0369610        13                  O Mundo dos Dinossauros     BR   
4   tt0369610        14                           Jurassic World     FR   
5   tt0369610        15                           Jurassic World     GR   
6   tt0369610        16                           Jurassic World     IT   
7   tt0369610        17                            Jurski svijet     HR   
8   tt0369610        18                             Olam ha'Yura     IL   
9   tt0369610        19           Jurassic World: Mundo Jurásico     MX   
10  tt0369610         1          Jurassic World: Sauruste maailm     EE   
11  tt0369610        20                           Jurassic World     SE   
12  tt0369610        21                        Jurassic World 3D     US   
13  tt0369610        22                        Jurassic World 3D     DE   
14  tt0369610        23                        Jurassic World 3D    XWW   
15  tt0369610        24                   Juras laikmeta pasaule     LV   
16  tt0369610        25                          Mundo Jurássico     PT   
17  tt0369610        26                      Мир Юрского периода     RU   
18  tt0369610        27                              Jurský svět     CZ   
19  tt0369610        28                         Monde jurassique     CA   

   language        types   attributes  is_original_title  
0        bg          NaN          NaN                0.0  
1       NaN  imdbDisplay          NaN                0.0  
2       NaN  imdbDisplay          NaN                0.0  
3       NaN          NaN  short title                0.0  
4       NaN  imdbDisplay          NaN                0.0  
5       NaN  imdbDisplay          NaN                0.0  
6       NaN  imdbDisplay          NaN                0.0  
7       NaN  imdbDisplay          NaN                0.0  
8        he  imdbDisplay          NaN                0.0  
9       NaN  imdbDisplay          NaN                0.0  
10      NaN  imdbDisplay          NaN                0.0  
11      NaN  imdbDisplay          NaN                0.0  
12      NaN          NaN  3-D version                0.0  
13      NaN          NaN  3-D version                0.0  
14      NaN          NaN  3-D version                0.0  
15      NaN  imdbDisplay          NaN                0.0  
16      NaN  imdbDisplay          NaN                0.0  
17      NaN          NaN          NaN                0.0  
18      NaN          NaN          NaN                0.0  
19       fr  imdbDisplay          NaN                0.0

In [32]:
imdb_title_akas_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 331703 entries, 0 to 331702
Data columns (total 8 columns):
title_id             331703 non-null object
ordering             331703 non-null int64
title                331703 non-null object
region               278410 non-null object
language             41715 non-null object
types                168447 non-null object
attributes           14925 non-null object
is_original_title    331678 non-null float64
dtypes: float64(1), int64(1), object(6)
memory usage: 20.2+ MB


In [33]:
imdb_title_akas_df.describe()

ordering  is_original_title
count  331703.000000      331678.000000
mean        5.125872           0.134769
std         6.706664           0.341477
min         1.000000           0.000000
25%         1.000000           0.000000
50%         2.000000           0.000000
75%         6.000000           0.000000
max        61.000000           1.000000

In [37]:
imdb_title_akas_df[imdb_title_akas_df['title_id'] == 'tt0369610']

title_id  ordering                                    title region  \
0   tt0369610        10                            Джурасик свят     BG   
1   tt0369610        11                        Jurashikku warudo     JP   
2   tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3   tt0369610        13                  O Mundo dos Dinossauros     BR   
4   tt0369610        14                           Jurassic World     FR   
5   tt0369610        15                           Jurassic World     GR   
6   tt0369610        16                           Jurassic World     IT   
7   tt0369610        17                            Jurski svijet     HR   
8   tt0369610        18                             Olam ha'Yura     IL   
9   tt0369610        19           Jurassic World: Mundo Jurásico     MX   
10  tt0369610         1          Jurassic World: Sauruste maailm     EE   
11  tt0369610        20                           Jurassic World     SE   
12  tt0369610        21                        Jurassic World 3D     US   
13  tt0369610        22                        Jurassic World 3D     DE   
14  tt0369610        23                        Jurassic World 3D    XWW   
15  tt0369610        24                   Juras laikmeta pasaule     LV   
16  tt0369610        25                          Mundo Jurássico     PT   
17  tt0369610        26                      Мир Юрского периода     RU   
18  tt0369610        27                              Jurský svět     CZ   
19  tt0369610        28                         Monde jurassique     CA   
20  tt0369610        29                           Jurassic World     US   
21  tt0369610         2                                 Ebb Tide     US   
22  tt0369610        30           Jurassic World: Mundo Jurásico     CL   
23  tt0369610        31                           Jurassic World     RO   
24  tt0369610        32                   Juros periodo pasaulis     LT   
25  tt0369610        33                           Jurassic World     FI   
26  tt0369610        34           Jurassic World: Mundo jurásico     PE   
27  tt0369610        35                          Jurassic Park 4     TR   
28  tt0369610        36                         Jurassic Park IV     US   
29  tt0369610        37                           Jurassic World     NO   
30  tt0369610        38                              Jurski svet     SI   
31  tt0369610        39                    Свiт Юрського перiоду     UA   
32  tt0369610         3                      The Gioi Khung Long     VN   
33  tt0369610        40                           Jurassic World     DK   
34  tt0369610        41                        Svet iz doba jure     RS   
35  tt0369610        42                              Jurský svet     SK   
36  tt0369610        43                           Jurassic World     TR   
37  tt0369610        44                          Jurassic Park 4     US   
38  tt0369610        45                           Jurassic World    NaN   
39  tt0369610        46                           Jurassic World     ES   
40  tt0369610        47                           Jurassic World     HU   
41  tt0369610        48                          Park jurajski 4     PL   
42  tt0369610         4                        Yur dövrü Dünyasi     AZ   
43  tt0369610         5                  Iuruli periodis samyaro     GE   
44  tt0369610         6                           Jurassic World     PL   
45  tt0369610         7                           Mundo Jurásico     UY   
46  tt0369610         8                        Jurassic World 3D     PL   
47  tt0369610         9          Jurassic World - Mundo Jurásico     AR   

   language        types                  attributes  is_original_title  
0        bg          NaN                         NaN                0.0  
1       NaN  imdbDisplay                         NaN                0.0  
2       NaN  imdbDisplay                         NaN                0.0  
3       NaN          NaN         

# imdb.title.basics.csv

## Summary

A table with movie id, two titles, release year, length and **genres**.

## columns
* tconst
  * the title key for the imdb tables
* primary_title 
  * ( there are 14,504 rows where the primary_title != the original_title)
* original_title
* start_year 
  * (presumably the release year)
* runtime_minutes
  * 31,739 NaN values
* genres
  * a list of genres ordered alphabetically, not by weight

## EDA

In [38]:
imdb_title_basics_df = pd.read_csv('datasets/imdb.title.basics.csv')

In [39]:
imdb_title_basics_df.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [40]:
imdb_title_basics_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
tconst             146144 non-null object
primary_title      146144 non-null object
original_title     146123 non-null object
start_year         146144 non-null int64
runtime_minutes    114405 non-null float64
genres             140736 non-null object
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [41]:
imdb_title_basics_df.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

In [43]:
imdb_title_basics_df[imdb_title_basics_df['runtime_minutes'].isna()]

tconst                                      primary_title  \
3       tt0069204                                    Sabse Bada Sukh   
6       tt0112502                                            Bigfoot   
8       tt0139613                                         O Silêncio   
16      tt0187902  How Huang Fei-hong Rescued the Orphan from the...   
21      tt0250404                                          Godfather   
...           ...                                                ...   
146131  tt9915790                                    Bobbyr Bondhura   
146138  tt9916428                                The Secret of China   
146140  tt9916622        Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                                    Dankyavar Danka   
146143  tt9916754                     Chico Albuquerque - Revelações   

                                           original_title  start_year  \
3                                         Sabse Bada Sukh        2018   
6                                                 Bigfoot        2017   
8                                              O Silêncio        2012   
16      How Huang Fei-hong Rescued the Orphan from the...        2011   
21                                              Godfather        2012   
...                                                   ...         ...   
146131                                    Bobbyr Bondhura        2019   
146138                                The Secret of China        2019   
146140        Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                                    Dankyavar Danka        2013   
146143                     Chico Albuquerque - Revelações        2013   

        runtime_minutes                 genres  
3                   NaN           Comedy,Drama  
6                   NaN        Horror,Thriller  
8                   NaN    Documentary,History  
16                  NaN                    NaN  
21                  NaN            Crime,Drama  
...                 ...                    ...  
146131              NaN                 Family  
146138              NaN  Adventure,History,War  
146140              NaN            Documentary  
146141              NaN                 Comedy  
146143              NaN            Documentary  

[31739 rows x 6 columns]

In [44]:
imdb_title_basics_df[imdb_title_basics_df['primary_title'] != imdb_title_basics_df['original_title']]

tconst                       primary_title  \
1       tt0066787     One Day Before the Rainy Season   
4       tt0100275            The Wandering Soap Opera   
11      tt0154039                So Much for Justice!   
13      tt0162942        Children of the Green Dragon   
15      tt0176694                  The Tragedy of Man   
...           ...                                 ...   
146026  tt9899938          Journey of the Sky Goddess   
146028  tt9900060  Lupin the Third: Fujiko Mine's Lie   
146037  tt9900688                   Big Three Dragons   
146121  tt9914254                       A Cherry Tale   
146135  tt9916170                       The Rehearsal   

                             original_title  start_year  runtime_minutes  \
1                           Ashad Ka Ek Din        2019            114.0   
4                     La Telenovela Errante        2017             80.0   
11                           Oda az igazság        2010            100.0   
13                 A zöld sárkány gyermekei        2010             89.0   
15                      Az ember tragédiája        2011            160.0   
...                                     ...         ...              ...   
146026                  Kibaiyanse! Watashi        2019            116.0   
146028  Lupin the IIIrd: Mine Fujiko no Uso        2019              NaN   
146037                          Da San Yuan        2019            111.0   
146121                    Kirsebæreventyret        2019             85.0   
146135                             O Ensaio        2019             51.0   

                         genres  
1               Biography,Drama  
4          Comedy,Drama,Fantasy  
11                      History  
13                        Drama  
15      Animation,Drama,History  
...                         ...  
146026             Comedy,Drama  
146028    Adventure,Crime,Drama  
146037                   Comedy  
146121              Documentary  
146135                    Drama  

[14504 rows x 6 columns]

# imdb.title.crew.csv

## Summary

A table with all the directors and writers for movies.

## Columns
* tconst
* directors
  * can be single entry, list, or NaN
* writers
  * can be single entry, list, or NaN

## EDA

In [46]:
imdb_title_crew_df = pd.read_csv('datasets/imdb.title.crew.csv')

In [47]:
imdb_title_crew_df.head()

tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

In [48]:
imdb_title_crew_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 3 columns):
tconst       146144 non-null object
directors    140417 non-null object
writers      110261 non-null object
dtypes: object(3)
memory usage: 3.3+ MB


# imdb.title.principals.csv

## Summary

A table of the principals on a film with the actors given their character names.  Everyone's actual names must be retrieved from imdb_name_basics_df and the film they are working on must be retrieved from imdb_title_basics_df.

## Columns
* tconst
* ordering
* nconst
* category
* job
  * mostly NaN
  * appear to be additional info for category, although there are plenty where both category and job are producer
* characters

## EDA

In [49]:
imdb_title_principals_df = pd.read_csv('datasets/imdb.title.principals.csv')

In [50]:
imdb_title_principals_df.head()

tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

In [51]:
imdb_title_principals_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1028186 entries, 0 to 1028185
Data columns (total 6 columns):
tconst        1028186 non-null object
ordering      1028186 non-null int64
nconst        1028186 non-null object
category      1028186 non-null object
job           177684 non-null object
characters    393360 non-null object
dtypes: int64(1), object(5)
memory usage: 47.1+ MB


In [57]:
imdb_title_principals_df[imdb_title_principals_df['job'].notna()]['category'].value_counts()

producer               113724
writer                  45494
director                 7504
cinematographer          7392
composer                 2003
editor                   1452
production_designer       115
Name: category, dtype: int64

In [126]:
imdb_title_principals_df['category'].value_counts()

actor                  256718
director               146393
actress                146208
producer               113724
cinematographer         80091
composer                77063
writer                  74357
self                    65424
editor                  55512
production_designer      9373
archive_footage          3307
archive_sound              16
Name: category, dtype: int64

# imdb.title.ratings.csv

## Summary

An average rating and number of ratings for each movie.

## Columns
* tconst
* averagerating
* numvotes

## EDA

In [60]:
imdb_title_ratings_df = pd.read_csv('datasets/imdb.title.ratings.csv')

In [61]:
imdb_title_ratings_df.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [62]:
imdb_title_ratings_df.describe()

averagerating      numvotes
count   73856.000000  7.385600e+04
mean        6.332729  3.523662e+03
std         1.474978  3.029402e+04
min         1.000000  5.000000e+00
25%         5.500000  1.400000e+01
50%         6.500000  4.900000e+01
75%         7.400000  2.820000e+02
max        10.000000  1.841066e+06

In [67]:
# what is the movie with the maximum votes (a major outlier)
imdb_title_ratings_df.iloc[imdb_title_ratings_df['numvotes'].idxmax()]

tconst           tt1375666
averagerating          8.8
numvotes           1841066
Name: 63498, dtype: object

In [68]:
# look up tconst in imdb_title_basics_df
imdb_title_basics_df[imdb_title_basics_df['tconst'] == 'tt1375666']

tconst primary_title original_title  start_year  runtime_minutes  \
7066  tt1375666     Inception      Inception        2010            148.0   

                       genres  
7066  Action,Adventure,Sci-Fi

# rt.movie_info.tsv

## Summary

Table with a lot of movie info, but no movie names and no other table to refer to.  

Independent of actual movies, the release date, box office income and rating are important information.

## Columns
* id
* synopsis
* rating
* genre
* director
* writer
* theater_date
* dvd_date
* currency
  * the currency for the value in box_office
  * 340 are \\$ and the rest are Nan, presumably they are all in \\$
* box_office
* runtime
  * a string at least containing the word minutes
* studio

## EDA

In [70]:
rt_movie_info_df = pd.read_csv('datasets/rt.movie_info.tsv', sep='\t')

In [71]:
rt_movie_info_df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [72]:
rt_movie_info_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
id              1560 non-null int64
synopsis        1498 non-null object
rating          1557 non-null object
genre           1552 non-null object
director        1361 non-null object
writer          1111 non-null object
theater_date    1201 non-null object
dvd_date        1201 non-null object
currency        340 non-null object
box_office      340 non-null object
runtime         1530 non-null object
studio          494 non-null object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [75]:
rt_movie_info_df['currency'].value_counts(dropna=False)

NaN    1220
$       340
Name: currency, dtype: int64

# rt.reviews.tsv

## Summary

A table with movie reviews, but no way to attribute them to a particular movie.

## Columns
* id
  * foreign_key identifying the movie in rt_movie_info_df
* review
* rating
  * may by NaN
* fresh
  * boolean type value either fresh(good) or rotten(bad)
* critic
* top_critic
  * boolean type value of 0 and 1, with 1 being a top critic
* publisher
* date

## EDA

In [79]:
rt_reviews_df = pd.read_csv('datasets/rt.reviews.tsv', sep='\t', encoding ='latin1')

In [80]:
rt_reviews_df.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [81]:
rt_reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
id            54432 non-null int64
review        48869 non-null object
rating        40915 non-null object
fresh         54432 non-null object
critic        51710 non-null object
top_critic    54432 non-null int64
publisher     54123 non-null object
date          54432 non-null object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [82]:
rt_reviews_df['fresh'].value_counts()

fresh     33035
rotten    21397
Name: fresh, dtype: int64

In [85]:
rt_reviews_df['top_critic'].value_counts(dropna=False)

0    41336
1    13096
Name: top_critic, dtype: int64

# tmdb.movies.csv

## Summary

A table with movie titles and review data.

## Columns
* Unnamed:
  * probably a residual index
* genre_ids
  * a list of ids referring to some other table to be located
* id
  * the movie id
* original_title
* popularity
  * ranges from 0.6 to 80.7
  * don't know what it refers to
* release_date
* title
* vote_average
* vote_count

## EDA

In [86]:
tmdb_movies_df = pd.read_csv('datasets/tmdb.movies.csv')

In [87]:
tmdb_movies_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [88]:
tmdb_movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
Unnamed: 0           26517 non-null int64
genre_ids            26517 non-null object
id                   26517 non-null int64
original_language    26517 non-null object
original_title       26517 non-null object
popularity           26517 non-null float64
release_date         26517 non-null object
title                26517 non-null object
vote_average         26517 non-null float64
vote_count           26517 non-null int64
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [89]:
tmdb_movies_df.describe()

Unnamed: 0             id    popularity  vote_average    vote_count
count  26517.00000   26517.000000  26517.000000  26517.000000  26517.000000
mean   13258.00000  295050.153260      3.130912      5.991281    194.224837
std     7654.94288  153661.615648      4.355229      1.852946    960.961095
min        0.00000      27.000000      0.600000      0.000000      1.000000
25%     6629.00000  157851.000000      0.600000      5.000000      2.000000
50%    13258.00000  309581.000000      1.374000      6.000000      5.000000
75%    19887.00000  419542.000000      3.694000      7.000000     28.000000
max    26516.00000  608444.000000     80.773000     10.000000  22186.000000

# tn.movie_budgets.csv

## Summary

A table with movie titles, production budgets and gross.

## Columns
* id (can be ignored)
* release_date
* movie
* production_budget
* domestic_gross
* worldwide_gross

## EDA

In [118]:
tn_movie_budgets_df = pd.read_csv('datasets/tn.movie_budgets.csv')

In [119]:
tn_movie_budgets_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [120]:
tn_movie_budgets_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
id                   5782 non-null int64
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null object
domestic_gross       5782 non-null object
worldwide_gross      5782 non-null object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [121]:
# need to convert production_budget, domestic_gross, worldwide_gross to int
cols = ['production_budget', 'domestic_gross', 'worldwide_gross']

# get rid of the dollar signs
clean_tn_movie_budgets_df = tn_movie_budgets_df.apply(
    lambda x: x.str.replace('$', '') if x.name in cols else x)

In [122]:
# get rid of the commas
clean_tn_movie_budgets_df = clean_tn_movie_budgets_df.apply(
    lambda x: x.str.replace(',', '') if x.name in cols else x)


In [123]:
# convert to int

clean_tn_movie_budgets_df[cols] = clean_tn_movie_budgets_df[cols].apply(pd.to_numeric, errors='coerce')

clean_tn_movie_budgets_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0          425000000       760507625       2776345279  
1          410600000       241063875       1045663875  
2          350000000        42762350        149762350  
3          330600000       459005868       1403013963  
4          317000000       620181382       1316721747

In [124]:
clean_tn_movie_budgets_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
id                   5782 non-null int64
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null int64
domestic_gross       5782 non-null int64
worldwide_gross      5782 non-null int64
dtypes: int64(4), object(2)
memory usage: 271.2+ KB


In [125]:
clean_tn_movie_budgets_df.describe()

id  production_budget  domestic_gross  worldwide_gross
count  5782.000000       5.782000e+03    5.782000e+03     5.782000e+03
mean     50.372363       3.158776e+07    4.187333e+07     9.148746e+07
std      28.821076       4.181208e+07    6.824060e+07     1.747200e+08
min       1.000000       1.100000e+03    0.000000e+00     0.000000e+00
25%      25.000000       5.000000e+06    1.429534e+06     4.125415e+06
50%      50.000000       1.700000e+07    1.722594e+07     2.798445e+07
75%      75.000000       4.000000e+07    5.234866e+07     9.764584e+07
max     100.000000       4.250000e+08    9.366622e+08     2.776345e+09

In [127]:
clean_tn_movie_budgets_df[clean_tn_movie_budgets_df['worldwide_gross'] == 0]

id  release_date                 movie  production_budget  \
194   95  Dec 31, 2020              Moonfall          150000000   
479   80  Dec 13, 2017                Bright           90000000   
480   81  Dec 31, 2019      Army of the Dead           90000000   
535   36  Feb 21, 2020      Call of the Wild           82000000   
670   71  Aug 30, 2019             PLAYMOBIL           75000000   
...   ..           ...                   ...                ...   
5761  62  Dec 31, 2014  Stories of Our Lives              15000   
5764  65  Dec 31, 2007           Tin Can Man              12000   
5771  72  May 19, 2015      Family Motocross              10000   
5777  78  Dec 31, 2018                Red 11               7000   
5780  81  Sep 29, 2015  A Plague So Pleasant               1400   

      domestic_gross  worldwide_gross  
194                0                0  
479                0                0  
480                0                0  
535                0                0  
670                0                0  
...              ...              ...  
5761               0                0  
5764               0                0  
5771               0                0  
5777               0                0  
5780               0                0  

[367 rows x 6 columns]

**There are 367 movies with a budget but no income**

# What we have so far

* a table maybe from **Box Office Mojo**
  * with a list of movies with incomplete data on domestic and foreign gross, 
* a set of related tables from the **IMDB** that are linked by title keys and name keys
  * a table of people with a name key that work on the industry with their positions and movies they worked on indicated by a title key
  * A list of the multiple names a movie has used internationaly with their title key
  * A table with a primary & original movie name, its release year, runtime and genre with the title key
  * A table with directors and writers of movies using their name keys and the title keys.  Data that probably could have been pulled from the table of people working in the industry.
  * A list of the principle people working on films, the job they had, and for actors the character's name indicated by their name key and linked to a movie by title key
  * A table of crowd sourced ratings including the average rating and the number of votes, linked to movies by the title key
* Two tables from **Rotten Tomatoes** that cannot be linked to specific movies by name, but are linked by a key
  * A table that includes a synopsis, MPAA rating, genre, director, writer, release date with d/m/y, dvd date with d/m/y, box office gross, runtime, and studio
  * A table of reviews linked to a movie by key with the writer, publisher, boolean values of good or bad, top critic or not, and published date
* A table from **The Movie Database (TMDB)** which has a API to potentially gather more data, and get a data dictionary
  * The table contains, genre id, title, popularity, vote average, and vote count
  * https://www.themoviedb.org/documentation/api?language=en-US
* A table potentially from **The Numbers** (mentioned above in the EDA of the BOM Table)
  * the table has movie titles, production budget, domestic gross and worldwide gross
  

Other information that might be useful
* awards
* Profit adjustments / influencers
  * adjusting for inflation
    * use the CPI
    * https://www.maa.org/press/periodicals/loci/joma/the-consumer-price-index-and-inflation-adjust-numbers-for-inflation
    * https://data.bls.gov/cgi-bin/surveymost?cu
    * https://towardsdatascience.com/the-what-and-why-of-inflation-adjustment-5eedb496e080
  * influence of major economic indicators
    * unemployment
    * available credit
* How do we define "doing the best" at the box office
  * https://www.newyorker.com/culture/richard-brody/what-it-takes-to-make-a-great-movie
  * Profit
  * Reviews and Awards
  * Influence on society (presumably positive)
* What has already been published
  * https://minimaxir.com/2018/07/imdb-data-analysis/
    * looks at the IMDB data and gives some definitions of the data
  * https://www.kaggle.com/PromptCloudHQ/imdb-data
    * a licensed version of the IMDB data
    * has 43 Kernels

Assumptions about the audience:
* Microsoft is a for profit corporation and will be interested in profits above all else
* Microsoft current entertainment division is Gaming with Xbox
* entering the movie industry, could maybe support the gaming division, and not the other way around given that gaming is said to be a larger industry than the movie industry now
* Microsoft did have a movie studio, XBOX Entertainment Studios, from 2012-2014 (should that be included or ignored)
  * https://en.wikipedia.org/wiki/Xbox_Entertainment_Studios
* their 2019 annual report https://www.microsoft.com/investor/reports/ar19/index.html